source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -88,29 +103,88 @@ GeoipProvider::~GeoipProvider() {
   ENVOY_LOG(debug, "Shutting down Maxmind geolocation provider");
   if (city_db_) {
     MMDB_close(city_db_.get());
+    city_db_ = nullptr;
   }
   if (isp_db_) {
     MMDB_close(isp_db_.get());
+    isp_db_ = nullptr;
   }
   if (anon_db_) {
     MMDB_close(anon_db_.get());
+    anon_db_ = nullptr;
+  }
+  if (country_mapping_) {
+    country_mapping_->clear();
+    country_mapping_ = nullptr;
   }
 }
 
 MaxmindDbPtr GeoipProvider::initMaxMindDb(const absl::optional<std::string>& db_path) {
   if (db_path) {
-    MMDB_s maxmind_db;
-    int result_code = MMDB_open(db_path.value().c_str(), MMDB_MODE_MMAP, &maxmind_db);
+    MMDB_s* maxmind_db = new MMDB_s;
```

Comment:
please revert to previous version, Envoy uses smart pointers for memory management, this is the reason you got a memory leak.